recipes/libnghttp2/all/conanfile.py

```diff
@@ -27,62 +27,94 @@ class Nghttp2Conan(ConanFile):
+
+    exports_sources = "src/*"
 
     def configure(self):
         if self.options.shared:
             self.options.rm_safe("fPIC")
-        if not (self.options.with_app or self.options.with_hpack or self.options.with_asio):
-            self.settings.rm_safe("compiler.cppstd")
-            self.settings.rm_safe("compiler.libcxx")
         if not self.options.with_app:
             del self.options.with_jemalloc
-        if Version(self.version) >= "1.52.0":
-            del self.options.with_asio
+        if not (self.options.with_app or self.options.with_hpack):
+            self.settings.rm_safe("compiler.cppstd")
+            self.settings.rm_safe("compiler.libcxx")
 
     def layout(self):
         cmake_layout(self, src_folder="src")
 
     def requirements(self):
-        if self.options.with_app or self.options.get_safe("with_asio"):
+        if self.options.with_app or self.options.with_hpack:
             self.requires("openssl/[>=1.1 <4]")
+            self.requires("zlib/[>=1.2.11 <2]")
         if self.options.with_app:
-            self.requires("c-ares/1.25.0")
             self.requires("libev/4.33")
-            self.requires("libevent/2.1.12")
-            self.requires("libxml2/2.12.4")
-            self.requires("zlib/[>=1.2.11 <2]")
+            self.requires("c-ares/[>=1.27 <2]")
+            self.requires("libxml2/[>=2.12.5 <3]")
+            self.requires("brotli/1.1.0")
             if self.options.with_jemalloc:
                 self.requires("jemalloc/5.3.0")
         if self.options.with_hpack:
             self.requires("jansson/2.14")
-        if self.options.get_safe("with_asio"):
-            self.requires("boost/1.84.0")
 
     def validate(self):
-        if self.options.get_safe("with_asio") and is_msvc(self):
-            raise ConanInvalidConfiguration("Build with asio and MSVC is not supported yet, see upstream bug #589")
         if self.settings.compiler == "gcc" and Version(self.settings.compiler.version) < "6":
             raise ConanInvalidConfiguration(f"{self.ref} requires GCC >= 6.0")
 
     def source(self):
-        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+        # get(self, **self.conan_data["sources"][self.version], strip_root=True)
+        pass
 
     def generate(self):
         tc = CMakeToolchain(self)
+
+        tc.cache_variables["BUILD_TESTING"] = False
+        tc.cache_variables["ENABLE_DOC"] = False
+        if self.options.with_app:
+            tc.cache_variables["LIBEV_FOUND"] = True
+            tc.cache_variables["LIBCARES_FOUND"] = True
+            tc.cache_variables["LIBBROTLIENC_FOUND"] = True
+            tc.cache_variables["LIBBROTLIDEC_FOUND"] = True
+            tc.cache_variables["LIBXML2_FOUND"] = True
+        if self.options.with_hpack:
+            tc.cache_variables["JANSSON_FOUND"] = True
+
+            # Disable Python finding so we don't build docs
+        tc.cache_variables["CMAKE_DISABLE_FIND_PACKAGE_Python3"] = True
```

Comment:
This one is NOT for testing and will remain after the PR leaves draft, it avoids cmake from trying to build the documentation